src/main/java/org/dice_research/opal/civet/metrics/DateFormat2.java

```diff
@@ -0,0 +1,223 @@
+			return checkRange(range);
+		} else if (issued.contains("-")) {
+			String[] range = issued.split("-");
+			return checkRange(range);
+		} else if (issued.contains("�")) {
+			String[] range = issued.split("�");
+			return checkRange(range);
+		} else {
+
+			if (issued.contains("GMT") && issued.split(" ").length == 6)
+				return 4;
+
+			for (String string : dateTimeFormat) {
+				SimpleDateFormat Format = new SimpleDateFormat(string);
+				try {
+					Format.parse(issued.trim());
+					return 4;
+
+				} catch (ParseException pe) {
+					continue;
+				}
+			}
+
+			for (String string : dateFormat) {
+				SimpleDateFormat Format = new SimpleDateFormat(string);
+				try {
+					Format.parse(issued.trim());
+					return 3;
+
+				} catch (ParseException pe) {
+					continue;
+				}
+			}
+
+			for (String string : yearFormat) {
+				SimpleDateFormat Format = new SimpleDateFormat(string);
+				try {
+					Format.parse(issued.trim());
+					return 2;
+
+				} catch (ParseException pe) {
+					continue;
+				}
+			}
+		}
+		return 1;
+	}
+
+	public static int checkRange(String[] range) {
+
+		String fromDate = "", toDate = "";
+		if (range.length == 2) {
+			fromDate = range[0].trim();
+			toDate = range[1].trim();
+		}
+		if (fromDate.contains("GMT") && fromDate.split(" ").length == 6 && toDate.contains("GMT")
+				&& toDate.split(" ").length == 6)
+			return 4;
+		for (String string : dateTimeFormat) {
+			SimpleDateFormat Format = new SimpleDateFormat(string);
+			try {
+				Format.parse(fromDate);
+				Format.parse(toDate);
+				return 5;
+
+			} catch (ParseException pe) {
+				continue;
+			}
+		}
+
+		for (String string : dateFormat) {
+			SimpleDateFormat Format = new SimpleDateFormat(string);
+			try {
+				Format.parse(fromDate);
+				Format.parse(toDate);
+				return 4;
+
+			} catch (ParseException pe) {
+				continue;
+			}
+		}
+
+		for (String string : yearFormat) {
+			SimpleDateFormat Format = new SimpleDateFormat(string);
+			try {
+				Format.parse(fromDate);
+				Format.parse(toDate);
+				return 3;
+
+			} catch (ParseException pe) {
+				continue;
+			}
+		}
+		return 1;
+	}
+
+	private static final Logger logger = LogManager.getLogger();
+	private static final String descriptions = "if the date is in the format: DD/MM/YYYY hh:mm:s GMT - DD/MM/YYYY hh:mm:s GMT then give 5 stars"
+			+ "if the date is in the format: YYYY.MM.DD hh:mm:s GMT then give 4 stars"
+			+ "if the date is in the format: YYYY.MM.DD then give 3 stars"
+			+ "if the date is in the format: YYYY then give 2 stars"
+			+ "if the date property is empty or have invalid date format then give 1 star";
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+		int result = 1;
+		logger.info("Processing dataset " + datasetUri);
+		Resource dataset = ResourceFactory.createResource(datasetUri);
+		Statement dataSetSentence = model.getRequiredProperty(dataset, RDF.type);
+		Resource dataSet = dataSetSentence.getSubject();
```

Comment:
There are duplicates. "DataSet" is same as "dataset"....but definitely you might have faced problems because you created a resource and did not connect it with the model. 

1. So create dataset like this at line 176--> Resource dataset = model.createResource(datasetURI); 
2. and remove Resource dataSet at line 178 
3. Replace all "dataSet" with dataset.

Reply:
Yes, exactly instead of model.createResource(), I was using ResourceFactory, now its working fine. Thanks for your comments :)